bin/run.sh

```diff
@@ -1,55 +1,94 @@
 #!/usr/bin/env bash
 
+shopt -s extglob
+
 usage() {
     printf '%s\n' "$*"
-    printf '%s\n' "Usage: ${0##*/} slug inputDir outputDir"
+    printf '%s\n' "Usage: ${0##*/} slug input_dir output_dir"
     exit 1
 }
 
-analyze() {
-    local opts=(
-        "--shell=bash"  
-        "--external-sources"
-        "--check-sourced"
-        "--norc"
-        "--format=json1"
-    )
-    (
-        cd "$inDir" >/dev/null || exit 1
-        shellcheck "${opts[@]}" "${filename}.sh"
-    )
+validate_input() {
+    if ! [[ -d "$in_dir" && -r "$in_dir" ]]; then
+        usage "Error: not a dir or not readable: ${in_dir@Q}"
+    fi
+    if ! [[ -d "$out_dir" && -w "$out_dir" ]]; then
+        usage "Error: not a dir or not writable: ${out_dir@Q}"
+    fi
+    local source="$in_dir/${snake_slug}.sh"
+    if ! [[ -f $source && -r $source ]]; then
+        usage "Error: not a file or not readable: ${source@Q}"
+    fi
 }
 
-process() {
-    # this reads stdin and emits to stdout
-    ruby bin/process-shellcheck-output.rb
+array_includes() {
+    local -n arr=$1
+    local elem=$2 e
+    for e in "${arr[@]}"; do [[ "$e" == "$elem" ]] && return; done
+    return 1
```

Comment:
Note you can use an implicit `return` here similar to the prior line. Though it's the same behavior either way.